Content.Client/Fax/UI/FaxBoundUi.cs

```diff
@@ -22,12 +30,47 @@ protected override void Open()
         _window.OpenCentered();
 
         _window.OnClose += Close;
+        _window.FileButtonPressed += OnFileButtonPressed;
         _window.CopyButtonPressed += OnCopyButtonPressed;
         _window.SendButtonPressed += OnSendButtonPressed;
         _window.RefreshButtonPressed += OnRefreshButtonPressed;
         _window.PeerSelected += OnPeerSelected;
     }
 
+    private async void OnFileButtonPressed()
+    {
+        if(_dialogIsOpen)
+            return;
+        _dialogIsOpen = true;
+        var filters = new FileDialogFilters(new FileDialogFilters.Group("txt"));
+        await using var file = await _fileDialogManager.OpenFile(filters);
+
+        if(_window == null)
```

Comment:
You can just set _dialogIsOpen straight after this as it's set to false everywhere else anyway.